package.json

```diff
@@ -41,7 +41,7 @@
         "danger": "^11.2.3",
         "eslint-plugin-jsdoc": "^44.2.7",
         "jsdom": "^17.0.0",
-        "prettier": "^2.1.1",
+        "prettier": "^3.0.0",
```

Comment:
Inspired by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/47110

Reply:
Can we split this into separate PR? (technically scope is different, DT package removal vs updating project own dependencies).
Also there is a work done by DT maintainers that could impact `prettier` usage already, so let's keep the bump in separate PR, otherwise looks good

Reply:
CI [fails](https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/5561587802/jobs/10159358187) if I remove typings but don't bump Prettier. I created https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66088 – maybe we can get that change in first and then remove the typings.

Reply:
The fix for the CI failure is to modify the packages which currently depend on `@types/prettier` to include `@types/prettier` in package.json, not touch this file. The only reason bumping prettier at the root is working is because it's a dep of the repo root itself and is being found via `node_modules`; if you were deprecating any other package, this would not work.

See the last half of this, which describes the problem you're seeing: https://github.com/DefinitelyTyped/DefinitelyTyped#removing-a-package

Reply:
Thanks for this info @jakebailey! I created https://github.com/microsoft/DefinitelyTyped-tools/pull/695 as a start.

Reply:
As a next step, I plan to make a PR that bumps `babel-plugin-tester` and `pretty-quick` and lists `prettier` (v3.0) as their dependency. Do these bumps need to be semver-major?

Reply:
> As a next step, I plan to make a PR that bumps babel-plugin-tester and pretty-quick and lists prettier (v3.0) as their dependency. Do these bumps need to be semver-major?

I checked the source packages (not the `@types/` ones) and both still depend on prettier 2. So you should change the types packages depend on `@types/prettier@2` instead of `prettier@3`. Because DT requires PRs to leave it in a passing state, you'll have to do that as part of this PR.

You'll have to add `@types/prettier` to https://github.com/microsoft/DefinitelyTyped-tools/blob/master/packages/definitions-parser/allowedPackageJsonDependencies.txt before that addition will pass CI.

Thanks for splitting out the DT upgrade to prettier to a separate PR. For this PR you should revert that change as @jakebailey pointed out.

Reply:
> They key problem I am trying to address is communication. Folks who upgrade from prettier@2 to prettier@3 should see that there is no more need to install @types/prettier. So we need to publish empty @types/prettier@3.0.0 and mark the package as deprecated.

Adding prettier to notNeededPackages.json will publish the empty `@types/pretter@3` and deprecate it.

Reply:
Thanks for your feedack @sandersn, @jakebailey and @peterblazejewicz! CI is passing now. I added both `prettier` and `@types/prettier` to `allowedPackageJsonDependencies.txt`:

- https://github.com/microsoft/DefinitelyTyped-tools/pull/695
- https://github.com/microsoft/DefinitelyTyped-tools/pull/696